File: JavaScript/CWE-78.js
const { exec } = require('child_process');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Введите директорию: ', (directory) => {
    // Уязвимая строка
    const command = `ls ${directory}`;
    exec(command, (error, stdout, stderr) => {
        if (error) {
            console.error(`Ошибка выполнения команды: ${error.message}`);
            return;
        }
        if (stderr) {
            console.error(`stderr: ${stderr}`);
            return;
        }
        console.log(`stdout:\n${stdout}`);
    });

    rl.close();
});

File: GO/CWE-78.go
package main

import (
	"fmt"
	"os/exec"
)

func listDirectory(directory string) {
	// Уязвимая строка
	cmd := exec.Command("sh", "-c", "ls "+directory)
	output, err := cmd.CombinedOutput()
	if err != nil {
		fmt.Println("Ошибка выполнения команды:", err)
		return
	}
	fmt.Println(string(output))
}

func main() {
	var userInput string
	fmt.Print("Введите директорию: ")
	fmt.Scanln(&userInput)
	listDirectory(userInput)
}
